index.js

```diff
@@ -18,22 +27,49 @@ function waitForActions(store, actions, timeout) {
         actions = actions.map((value) => typeof value === 'string' ? { type: value } : value);
     }
 
-    const shouldPromiseResolve = typeof actions === 'function' ?
-        actions :
-        (storeActions) => actionsMatch(actions, storeActions);
+    let cancel = () => {};
 
-    // If the store already contains the expected actions, resolve the Promise immediately
-    if (shouldPromiseResolve(store.getActions())) {
-        const promise = Promise.resolve();
+    const promise = new Promise((resolve, reject) => {
+        const actionsMatch = (storeActions) => {
+            if (typeof actions === 'function') {
```

Comment:
This part of the diff until line 71 can be simplified a bit, will talk to you personally.